lib/session.js

```diff
@@ -414,9 +417,12 @@ Session.prototype._handleDisposition = function(frame) {
     state: frame.state
   };
 
-  _.values(this._linksByName).forEach(function(link) {
-    link._dispositionReceived(disposition);
-  });
+  var dispositionHandler = function(l) { l._dispositionReceived(disposition); };
+  if (frame.role === constants.linkRole.sender) {
+    this._senderLinks.forEach(dispositionHandler);
```

Comment:
oh, indeed you are correct. The spec is very mysterious in their wording here, but it should have been obvious from the test that the sender should be the one receiving dispositions!